src/main/java/uk/ac/ebi/embl/api/validation/helper/MasterSourceFeatureUtils.java

```diff
@@ -112,7 +112,18 @@ public SourceFeature constructSourceFeature(SampleEntity sample, TaxonHelper tax
 		sourceFeature.setTaxId(sampleInfo.getTaxId());
 		sourceFeature.setScientificName(sampleInfo.getScientificName());
 		sourceFeature.setMasterLocation();
+		addQualifiers(sourceFeature,sample,taxonHelper,sampleInfo);
+		return sourceFeature;
+	}
 
+	public SourceFeature updateSourceFeature(SourceFeature sourceFeature, SampleEntity sample, TaxonHelper taxonHelper, SampleInfo sampleInfo) {
```

Comment:
- Change class name to SourceFeatureUtils because it is not only used for Masters but also for template sequences.